kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/interceptors/TimeoutInterceptor.kt

```diff
@@ -32,19 +46,68 @@ internal class TimeoutInterceptor(
       return when (val timeout = testCase.config.timeout) {
          null -> test(testCase, scope)
          else -> try {
-            withTimeout(timeout) {
+            withAppropriateTimeout(timeout) {
                test(testCase, scope.withCoroutineContext(coroutineContext))
             }
-         } catch (t: Throwable) {
-            logger.log { Pair(testCase.name.testName, "Caught timeout $t") }
-            TestResult.Error(mark.elapsedNow(), TestTimeoutException(timeout, testCase.name.testName))
+         } catch (t: CancellationException) {
```

Comment:
Was there some reason to catch `Throwable` instead of a more specific exception type like `CancellationException` or `TimeoutCancellationException`?

Reply:
Catching `Throwable` seems like a mistake to me. Shouldn't we be even more specific, catching just `TimeoutCancellationException` and `KotestTimeoutCancellationException` (which might be named `WallclockTimeoutCancellationException` to better communicate intent)?

Reply:
OK, I see that narrowing the Cancellation types has been done inside the catch block.